tools/orca/src/run.rs

```diff
@@ -0,0 +1,162 @@
+use crate::error::Error;
+use crate::state::*;
+
+use std::sync::Arc;
+
+use rand::seq::SliceRandom;
+use rand::SeedableRng;
+use rand_chacha::ChaCha8Rng;
+
+use crossbeam::queue::SegQueue;
+
+use kanidm_client::{KanidmClient, KanidmClientBuilder};
+
+use tokio::sync::broadcast;
+
+use std::time::{Duration, Instant};
+
+async fn actor_person(
+    client: KanidmClient,
+    person: Person,
+    stats_queue: Arc<SegQueue<EventRecord>>,
+) -> Result<(), Error> {
+    let model = person.model.into_dyn_object();
+
+
+    // let transition, delay = determine_next_state
+
+    // From the person, what model did they request?
+
+    // let next_state =
+
+    // Setup their initial state from that model.
+
+    // Execute any delay, loop
+
+    // Select on the broadcast too.
+
+    // Submit stats as we go.
+
+    Ok(())
+}
+
+pub enum EventRecord {}
+
+#[derive(Clone, Debug)]
+pub enum Signal {
+    Stop,
+}
+
+async fn execute_inner(
+    warmup: Duration,
+    test_time: Option<Duration>,
+    mut control_rx: broadcast::Receiver<Signal>,
+) -> Result<(Instant, Instant), Error> {
+    // Delay for warmup time.
+    // TODO: Read warmup time from profile.
```

Comment:
Can't we pass it as a parameter and read it inside the calling function `execute` since we already have it there?

Reply:
Well yeah, I just hadn't added it yet.